### ISSUE TITLE
Confusing examples in "Introduction to PyTorch for former Torchies.ipynb"

### ISSUE THREAD

**Issue by blackyang** [pytorch/tutorials#38]
Regarding to the effects of "retain_variables" argument, the notebook says:

```
# this would fail if we didn't specify 
# that we want to retain variables
```

however when I set `retain_variables=False` in `y.backward()`, it seems that nothing wrong happens. I can still get `x.grad`

**Comment by soumith** [comment]
thanks for checking on this.

This particular example doesn't need any forward buffers to compute it's backward gradients, and hence works wven without `retain_variables=True`. However, if you take a case like `y = x^2` which needs to hold on to forward value to compute backward, it will fail

**Comment by blackyang** [comment]
@soumith thanks for your reply. Yes this example doesn't need to store forward values, I just made a simple pull request to fix this issue #40
